ivadomed/main.py

```diff
@@ -318,20 +318,34 @@ def run_command(context, n_gif=0, thr_increment=None, resume_training=False):
 
         options = None
         for subject in bids_subjects:
-            fname_img = subject.record["absolute_path"]
+            if context['loader_parameters']['multichannel']:
+                fname_img = []
+                contrasts = context['loader_parameters']['contrast_params']['testing']
+                # Keep contrast order
+                for c in contrasts:
+                    for s in bids_subjects:
+                        if subject.record['subject_id'] == s.record['subject_id'] and s.record['modality'] == c:
+                            fname_img.append(s.record['absolute_path'])
+                            bids_subjects.remove(s)
```

Comment:
Do you think we should check here if a given subject has all requested contrasts? To avoid adding a subject with missing contrast. It would be detected further down the line I would say but.. the sooner the better?

Reply:
Yes good idea, I'll add it :)

Reply:
Added it here : 6ecb85d